airflow/cli/cli_config.py

```diff
@@ -90,7 +91,7 @@ def _check_value(self, action, value):
                         f"executors derived from them, your current executor: {executor}, subclassed from: "
                         f'{", ".join([base_cls.__qualname__ for base_cls in executor_cls.__bases__])}'
                     )
-                    raise ArgumentError(action, message)
+                    warnings.warn(message)
```

Comment:
I think this one is way better - it issues warning, while allowing you to still run `airflow celery worker --help` and see the options even if you do not have CeleryExecutor configured yet.

Reply:
Most of this code will leave soon anyways when I finish the CLI abstraction for Airflow CLI (see our [convo here](https://github.com/apache/airflow/pull/29055#discussion_r1082091328)).